webapp/components/compat-2021.js

```diff
@@ -254,3 +255,228 @@ class Compat2021Summary extends PolymerElement {
+          <option value="aspect-ratio">aspect-ratio</option>
+          <option value="css-flexbox">css-flexbox</option>
+          <option value="css-grid">css-grid</option>
+          <option value="css-transforms">css-transforms</option>
+          <option value="position-sticky">position-sticky</option>
+        </select>
+      </div>
+
+      <!-- TODO: replace with google-chart polymer element? -->
+      <div id="failuresChart" class="chart"></div>
+`;
+  }
+
+  static get properties() {
+    return {
+      stable: Boolean,
+      feature: String,
+      chartOptions: {
+        type: Object,
+        readyOnly: true,
+        value: {
+          width: 800,
+          height: 350,
+          chartArea: {
+            height: '80%',
+          },
+          tooltip: {
+            trigger: 'both',
+          },
+          hAxis: {
+            title: 'Date',
+            format: 'MMM-YYYY',
+          },
+          vAxis: {
+            title: 'Percentage of passing tests',
+            format: 'percent',
+            viewWindow: {
+              // We set a global minimum value for the y-axis to keep the graphs
+              // consistent when you switch features. Currently the lowest value
+              // is aspect-ratio, with a ~25% pass-rate on Safari STP, Safari
+              // Stable, and Firefox Stable.
+              min: 0.2,
+              max: 1,
+            }
+          },
+          explorer: {
+            actions: ['dragToZoom', 'rightClickToReset'],
+            axis: 'horizontal',
+            keepInBounds: true,
+            maxZoomIn: 4.0,
+          },
+          colors: ['#4285f4', '#ea4335', '#fbbc04'],
+        }
+      },
+    };
+  }
+
+  static get observers() {
+    return [
+      'updateChart(feature, stable)',
+    ];
+  }
+
+  static get is() {
+    return 'compat-2021-feature-chart';
+  }
+
+  ready() {
+    super.ready();
+    const params = (new URL(document.location)).searchParams;
+    this.feature = params.get('feature');
+
+    // The default behavior of the page (when loaded with no params) is to not
+    // select any graph, so we can directly set `value` from the param here.
+    this.$.featureSelect.value = this.feature;
+
+    this.$.featureSelect.addEventListener('change', () => {
+      this.feature = this.$.featureSelect.value;
+    });
+  }
+
+  async updateChart(feature, stable) {
+    // Our observer may be called before the feature is set, so debounce that.
+    if (!feature) {
+      return;
+    }
+
+    // Ensure that Google Charts has loaded.
+    await load();
+
+    const div = this.$.failuresChart;
+    const label = stable ? 'stable' : 'experimental';
+    const url = `${GITHUB_URL_PREFIX}/data/compat2021/${feature}-${label}.csv`;
+
+    const csvResp = await fetch(url);
+    if (!csvResp.ok) {
+      throw new Error(`Fetching chart csv data failed: ${csvResp.status}`);
+    }
+    const csvText = await csvResp.text();
+    const csvLines = csvText.split('\n');
```

Comment:
This code looks very familiar. Will there be more, or just one dupe? If it's trivial to factor out a `fetchCSV()` that might be an improvement even with only two call sites.